packages/block-editor/src/components/block-list/style.scss

```diff
@@ -486,9 +486,83 @@
 	background-color: $white;
 
 	.block-editor-block-toolbar .components-toolbar-group,
-	.block-editor-block-toolbar .components-toolbar {
+	.block-editor-block-toolbar .components-toolbar,
+	.block-editor-block-toolbar__mover-switcher-container {
 		border-right-color: $dark-gray-primary;
 	}
+
+	.block-editor-block-toolbar__mover-switcher-container {
+		border-right-style: solid;
+		border-right-width: $border-width;
+	}
+
+	// Adjust the mover control to fit as a unit next to the block switcher.
+	.block-editor-block-toolbar__block-switcher-wrapper {
+		// Adjust the focus rectangle for the switcher.
+		.block-editor-block-switcher__no-switcher-icon::before,
+		.block-editor-block-switcher__toggle::before {
+			min-width: 0;
+			width: 28px;
```

Comment:
there are some random numbers here. Are these based on any variables?

Reply:
Sort of. The focus rectangles are 32x32 ($grid-unit-40) inside 48x48 buttons ($grid-unit-60), but the focus rectangle has been massaged to better fit the mover control on the right:

<img width="182" alt="Screenshot 2020-05-29 at 11 24 36" src="https://user-images.githubusercontent.com/1204802/83243983-01c85680-a19f-11ea-8ef4-41c197ee73cf.png">

A variable could be `$grid-unit-60 - $grid-unit-10 - $grid-unit-15` = 28 instead, which is accurate here. But I'll see if I can refactor this a bit better.